deepchem/feat/graph_features.py

```diff
@@ -676,6 +693,36 @@ def __init__(self, master_atom=False, use_chirality=False,
     self.master_atom = master_atom
     self.use_chirality = use_chirality
     self.atom_properties = list(atom_properties)
+    self.per_atom_fragmentation = per_atom_fragmentation
+
+  def featurize(self, molecules, log_every_n=1000) -> np.ndarray:
+    """
+    Override parent: aim is to add handling atom-depleted molecules featurization
+    
+    Parameters
+    ----------
+    molecules: rdkit.Chem.rdchem.Mol / SMILES string / iterable
+      RDKit Mol, or SMILES string or iterable sequence of RDKit mols/SMILES
+      strings.
+    log_every_n: int, default 1000
+      Logging messages reported every `log_every_n` samples.
+
+    Returns
+    -------
+    features: np.ndarray
+      A numpy array containing a featurized representation of `datapoints`.
+    """
+    features = super(ConvMolFeaturizer, self).featurize(
+        molecules, log_every_n=1000)
+    if self.per_atom_fragmentation:
+      valid_frag_inds = [[
```

Comment:
Can you add a comment explaining how this work? I think `features` above would be an array of `ConvMol` objects. How can we expand that out in a double for loop? My apologies, I'm likely missing something basic here

Reply:
> 
> 
> Can you add a comment explaining how this work? I think `features` above would be an array of `ConvMol` objects. How can we expand that out in a double for loop? My apologies, I'm likely missing something basic here
Yes, it's hard to get it, i wrote a commnet in code
I  create temporary _valid ids_ seving to filter out failed featurizations from every sublist 
of features (i.e. every molecules' frags list), and also to filter out  totally failed sublists.
This makes output digestable by Loaders, otherwise we would need some changes in them , what you dont want (you suggested rather to override `featurize()`)